juju/juju
# discoverspaces: Get subnets when provider doesn't support space discovery

## Description of change

At the moment we only get subnets from the provider if it supports space discovery, but it would be useful to have them for providers that support networking (even if we don't have them tagged with a space). That will allow us to use them to set up ingress rules when a cross-model relation is created.

Change the discover spaces worker to collect subnets (with no associated spaces) if the provider doesn't support space discovery. Also change the AddSubnets API call to allow subnets without a space, and change the ListSubnets API method and command to handle empty space.

Rewrite the worker tests to not use `JujuConnSuite`.

## QA steps

* Bootstrap on AWS. 
* Run `juju subnets` to verify that subnets have been imported in by the discoverspaces worker - they'll come back with a blank space tag.
